river/utils/random.py

```diff
@@ -27,3 +30,38 @@ def poisson(rate: float, rng=random) -> int:
         p *= rng.random()
 
     return k - 1
+
+
+def sample_discrete(distribution: list[float]) -> int:
```

Comment:
Does [`random.choices`](https://docs.python.org/3/library/random.html#random.choices) not suit what you need?

Reply:
Thanks, I removed the "sample_discrete" function in latest commit and used "random.choices" instead.